crates/matrix-sdk-crypto/src/backups/keys/backup.rs

```diff
@@ -90,6 +194,119 @@ impl MegolmV1BackupKey {
         self.inner.version.lock().unwrap().clone()
     }
 
+    /// Get the MAC key that is used with the backup key
+    pub(crate) fn mac_key(&self) -> Option<&HmacSha256Key> {
+        self.inner.mac_key.as_ref()
+    }
+
+    /// Set the backup version that this `MegolmV1BackupKey` will be used with.
+    ///
+    /// The key won't be able to encrypt room keys unless a version has been
+    /// set.
+    pub fn set_version(&self, version: String) {
+        *self.inner.version.lock().unwrap() = Some(version);
+    }
+
+    pub(crate) async fn encrypt(&self, user_id: &UserId, session: InboundGroupSession) -> KeyBackupData {
```

Comment:
Should we be passing the user ID here, or should `MegolmV1BackupKey` contain the user ID?  `MegolmV1BackupKey` is created by `BackupDecryptionKey`, which does not have the user ID, but `encrypt` is called by something that has the user ID.